repos/fdbt-site/src/pages/manageFareDayEnd.tsx

```diff
@@ -0,0 +1,114 @@
+import ErrorSummary from '../components/ErrorSummary';
+import React, { ReactElement, useState } from 'react';
+import CsrfForm from '../components/CsrfForm';
+import { BaseLayout } from '../layout/Layout';
+import { ErrorInfo, NextPageContextWithSession } from '../interfaces';
+import FormElementWrapper, { FormGroupWrapper } from '../components/FormElementWrapper';
+import { getSessionAttribute, updateSessionAttribute } from '../utils/sessions';
+import { GS_FARE_DAY_END_ATTRIBUTE } from '../constants/attributes';
+import { getAndValidateNoc, getCsrfToken } from '../utils';
+import { getFareDayEnd } from '../data/auroradb';
+import { extractGlobalSettingsReferer } from '../utils/globalSettings';
+import SubNavigation from '../layout/SubNavigation';
+import InfoPopup from '../components/InfoPopup';
+
+const title = 'Manage Fare Day End - Create Fares Data Service';
+const description = 'Manage Fare Day End page of the Create Fares Data Service';
+
+export const fareDayEndInputId = 'fare-day-end-input';
+
+type ManageFareDayEndProps = {
+    errors: ErrorInfo[];
+    csrfToken: string;
+    fareDayEnd: string;
+    referer: string | null;
+    saved: boolean;
+};
+
+const ManageFareDayEnd = ({ errors, csrfToken, fareDayEnd, referer, saved }: ManageFareDayEndProps): ReactElement => {
+    const [showSaved, setShowSaved] = useState(saved);
+
+    return (
+        <BaseLayout title={title} description={description} showNavigation referer={referer}>
+            <div className="govuk-width-container">
+                <div className="govuk-grid-row">
+                    <div className="govuk-grid-column-one-third">
+                        <SubNavigation />
+                    </div>
+
+                    <div className="govuk-grid-column-two-thirds">
+                        <ErrorSummary errors={errors} />
+
+                        <h1 className="govuk-heading-xl">Fare day end</h1>
+                        <p className="govuk-body govuk-!-margin-bottom-4" id={'fare-day-text'}>
+                            If your fare day extends past midnight, enter the time on the following morning when your
+                            tickets expire
+                        </p>
+
+                        <div className="govuk-inset-text">
+                            Enter times in 24hr format. For example 0900 is 9am, 1730 is 5:30pm.
+                        </div>
+
+                        <CsrfForm action="/api/manageFareDayEnd" method="post" csrfToken={csrfToken}>
+                            <FormGroupWrapper errorIds={[fareDayEndInputId]} errors={errors}>
+                                <fieldset className="govuk-fieldset">
+                                    <FormElementWrapper
+                                        errors={errors}
+                                        errorId={fareDayEndInputId}
+                                        errorClass="govuk-input--error"
```

Comment:
Using this page, it wasn't very obvious to me that I would be clearing a saved fare day end time if I clicked save after emptying the text box. Its also a little strange that upon first load, and when adding one, if I click save when not adding anything, it says it has saved. I think we either need to have some other functionality (clear button?) or some hint text to explain that they can clear it by emptying the box.

Reply:
Good shout, will discuss in standup tomorrow 👍

Reply:
Done